app/gui/src/controller/searcher/component/builder.rs

```diff
@@ -63,12 +63,24 @@ impl ModuleGroups {
 ///
 /// The builder allow extending the list with new entries, and build a list with properly sorted
 /// groups.
+///
+/// To build a [`component::List`] with non-empty [`component::List::favorites`], the following
+/// methods should be called:
+///  - [`set_favorites`] to specify order and grouping of [`Component`]s in favorites;
+///  - [`extend`] to specify IDs of [`Component`]s which should be retained from the initial
+///    [`component::Group`]s specified through [`set_favorites`] (allows filtering of favorites e.g.
+///    by self type);
```

Comment:
I don't understand these docs in several places:
1. they tell "to build component list with non-empty favourites ..." - so if I dont use these steps I would have empty favourites? 
2. `[`extend`] to specify IDs of [`Component`]s which should be retained from the initial [`component::Group`]s specified through [`set_favorites`]` – I was reading it few times and I dont understand it. It's a super complex sentence and there are multiple questions involved: What do you mean by "retained"? What do you mean by "specify IDs of components [...] specified trough 'set_favorites'? Is 'set_favorites' specifying components? Does the function "extend" allows "filtering favourites"?